core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/controlBlock/ifStatement/IfOpExec.scala

```diff
@@ -0,0 +1,23 @@
+package edu.uci.ics.texera.workflow.operators.controlBlock.ifStatement
+
+import edu.uci.ics.amber.engine.common.executor.OperatorExecutor
+import edu.uci.ics.amber.engine.common.model.State
+import edu.uci.ics.amber.engine.common.model.tuple.{Tuple, TupleLike}
+import edu.uci.ics.amber.engine.common.workflow.PortIdentity
+
+class IfOpExec(conditionName: String) extends OperatorExecutor {
+  private var output_port: Int = _
+
+  override def processState(state: State, port: Int): Option[State] = {
+    output_port = if (state.get(conditionName).asInstanceOf[Boolean]) 0 else 1
+    None
+  }
```

Comment:
I have tons of questions on this. You definitely need more explanation on this function. Please for sure to add more doc.

- For example, does this function expect 1 state, or it can handle multiple states? If it should not be invoked multiple times, add an assertion to prevent it.

- does each state must to have exactly 1 key-value pair, or is it fine to have multiple key-value pairs?  what happens for those other states not identified by `conditionName`?

- Do you expect a boolean value only? can it be other types? what happens if other types are passed? for example, if I pass a 1 int, can it be valid? should it be an error?

- What happens if there are other states coming, or other key-value pairs in the same state (not to be used by this if, but to be passed downstream)? for example "for operator" would send a "i" state. this "if operator" seems to eat this i without passing it down.

Reply:
1. This function can handle one or more states.
2. It can have mutiple key-value pairs. Keys are not identified `conditionName` will be ignored.
3. It can accept any value that can be converted to a boolean. For example, Int 1 will be converted to true.
4. I just made change that this function will pass the original state to downstream.

Those answers are documented on the code.